tests/integration/ha_tests/test_ha.py

```diff
@@ -65,6 +83,128 @@ async def test_build_and_deploy(ops_test: OpsTest) -> None:
+
+    # find leader unit
+    leader_unit = await find_unit(ops_test, leader=True)
+
+    # verify that we have a leader
+    assert leader_unit is not None, "No unit is leader"
+
+    # Force delete the leader and scale down
+    await kubectl_delete(ops_test, leader_unit, False)
+    await scale_application(ops_test, app, expected_units)
+
+    # grab unit hosts
+    hostnames = await get_units_hostnames(ops_test)
+
+    # check that the replica set with the remaining units has a primary
+    primary = await get_replica_set_primary(ops_test)
+
+    # verify that the primary is not None
+    assert primary is not None, "replica set has no primary"
+
+    # check that the primary is one of the remaining units
+    assert (
+        f"{primary.replace('/', '-')}.mongodb-k8s-endpoints" in hostnames
+    ), "replica set primary is not one of the available units"
+
+    # verify that the configuration of mongodb no longer has the deleted ip
+    member_hosts = await fetch_replica_set_members(ops_test)
+
+    # verify that the replica set members have the correct units
+    assert set(member_hosts) == set(hostnames), "mongod config contains deleted units"
+
+    # verify that the no writes were skipped
+    total_expected_writes = await get_total_writes(ops_test)
+    with await get_mongo_client(ops_test) as client:
+        actual_writes = client[TEST_DB][TEST_COLLECTION].count_documents({})
+    assert total_expected_writes == actual_writes, "writes to the db were missed."
+
+
+async def test_replication_across_members(ops_test: OpsTest, continuous_writes) -> None:
+    """Check consistency, ie write to primary, read data from secondaries."""
+    # first find primary, write to primary, then read from each unit
+    await insert_focal_to_cluster(ops_test)
+
+    primary = await get_replica_set_primary(ops_test)
+    app = await get_application_name(ops_test, APP_NAME)
+    excluded = [primary]
+    for unit in ops_test.model.applications[app].units:
+        if unit.name == primary:
+            continue
+        with await get_mongo_client(ops_test, excluded=excluded) as client:
+            db = client["new-db"]
+            test_collection = db["test_ubuntu_collection"]
+            query = test_collection.find({}, {"release_name": 1})
+            assert query[0]["release_name"] == "Focal Fossa"
+        excluded.append(unit.name)
+
+    # verify that the no writes were skipped
+    total_expected_writes = await get_total_writes(ops_test)
+    with await get_mongo_client(ops_test) as client:
+        actual_writes = client[TEST_DB][TEST_COLLECTION].count_documents({})
+    assert total_expected_writes == actual_writes
+
+
+async def test_replication_member_scaling(ops_test: OpsTest, continuous_writes) -> None:
+    """Verify newly added and newly removed members properly replica data.
+
+    Verify newly members have replicated data and newly removed members are gone without data.
+    """
```

Comment:
I think you should just add this as a check to the end of `test_scale_down_capablities` and `test_scale_up_capablities` rather than having an additional test here

Maybe you can creat a helper function for:
```

    with await get_mongo_client(ops_test, excluded=original_units) as client:
        db = client["new-db"]
        test_collection = db["test_ubuntu_collection"]
        query = test_collection.find({}, {"release_name": 1})
        assert query[0]["release_name"] == "Focal Fossa"

        # verify that the no writes were skipped
        total_expected_writes = await get_total_writes(ops_test)
        actual_writes = client[TEST_DB][TEST_COLLECTION].count_documents({})
        assert total_expected_writes == actual_writes
```